src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,38 @@
 package mate.academy;
 
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private final List<EventListener> listeners = new CopyOnWriteArrayList<>();
```

Comment:
Using CopyOnWriteArrayList ensures thread safety for add and remove operations, but it may not be the most efficient choice if the listeners list is frequently modified, as it creates a new copy of the list for every write operation. If modifications are infrequent compared to read operations, then this is a good choice.